File: src/main/java/com/lannstark/lec07/Lec07Main.java
package com.lannstark.lec07;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

public class Lec07Main {

  public static void main(String[] args) throws IOException {
      JavaFilePrinter printer = new JavaFilePrinter();
      printer.readFile();
  }

  private int parseIntOrThrow(@NotNull String str) {
    try {
      return Integer.parseInt(str);
    } catch (NumberFormatException e) {
      throw new IllegalArgumentException(String.format("%s is not a number", str));
    }
  }

  private Integer parseIntOrThrow2(@NotNull String str) {
    try {
      return Integer.parseInt(str);
    } catch (NumberFormatException e) {
      return null;
    }
  }

}

File: src/main/kotlin/lec7예외/Lec07Main.kt
package lec7예외

fun main () {

    val printer = FilePrinter()
    printer.readFile()

}

fun parseIntOrThrow(str: String): Int {
    try {
        return str.toInt()
    } catch (e: NumberFormatException) {
        throw IllegalArgumentException("$str is not a number.")
    }
}

fun parseIntOrThrow2(str: String): Int? {
    return try {
        str.toInt()
    } catch (e : NumberFormatException) {
        null
    } // try catch 구문 역시 expression이라 한번에 return 가능함
}